api/src/main/java/org/openmrs/module/appointments/service/impl/RecurringAppointmentServiceImpl.java

```diff
@@ -108,7 +107,7 @@ public void setEditAppointmentValidators(List<AppointmentValidator> editAppointm
     public void changeStatus(Appointment appointment, String status, Date onDate, String clientTimeZone) {
         List<String> errors = new ArrayList<>();
         AppointmentStatus appointmentStatus = AppointmentStatus.valueOf(status);
-        validateStatusChange(appointment, appointmentStatus, errors);
+        appointmentServiceHelper.validateStatusChange(appointment, appointmentStatus, errors, statusChangeValidators);
         if (errors.isEmpty()) {
```

Comment:
Can be extracted. Since this check is abstracted for validateAndSave and validateAndUpdate